app/src/main/java/io/appium/uiautomator2/utils/Device.java

```diff
@@ -36,11 +36,23 @@ public static void wake() throws RemoteException {
         getUiDevice().wakeUp();
     }
 
-    public static void scrollTo(String scrollToString) throws UiObjectNotFoundException {
-        // TODO This logic needs to be changed according to the request body from the Driver
+    public static void scrollToElement(UiSelector selector, int maxSwipes)
+            throws UiObjectNotFoundException {
         UiScrollable uiScrollable = new UiScrollable(new UiSelector().scrollable(true).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().descriptionContains(scrollToString).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().textContains(scrollToString).instance(0));
+        String uiScrollableClassName = uiScrollable.getClassName();
+        String hScrollViewClassName = android.widget.HorizontalScrollView.class.getName();
+
+        if(uiScrollableClassName != null && uiScrollableClassName.equals(hScrollViewClassName)) {
```

Comment:
missing space after if

Reply:
Fixed this one and another similar issue a few lines below.